### PR TITLE
Support externalized source maps, bundling customization

### PR DESCRIPTION
**Pull request by kwesterfeld**
Various fixes/enhancements, #14, #17, #20.   Let me know if you want squashed commits or each change baselined to master.

### CODE REVIEW

File: lib/handlers.js [hunk 1]
```diff
@@ -22,10 +22,42 @@ export const makeHandlers = ({_, config}) => {
...
+      if (builder) {
+        const options = merge(config.builder.options, builder.options);
+        if (req.originalUrl.endsWith('.map')) {
+          _.log(`requesting ${req.url} retrieving external sourcemap`)
+          _.bundle(builder.expression, options).then(({sourceMap}) => res.end(sourceMap))
+        } else {
+          _.log(`requesting ${req.url} triggering bundling`)
+          _.bundle(builder.expression, options).then(({source, sourceMap}) => res.end(
+            options.sourceMaps === 'inline' || !sourceMap
+              ? source
```
Comment (alexisvincent) [inline lib/handlers.js]: Not sure how much I'm a fan of building this in here... But it's ok for now. Might rework it in december.

File: lib/index.js [hunk 2]
```diff
@@ -240,11 +240,23 @@ const init = (configOverrides = {}) => {
   // f: notify system that a file has changed
   _.fileChanged = (absolutePath) => {
     _.log(`file-changed: ${path.relative(config.directories.root, absolutePath)}`);
+
```
Comment (alexisvincent) [inline lib/index.js]: Hmmmm. Why are you busting the cache here?

Check out https://github.com/alexisvincent/systemjs-tools/blob/master/lib/index.js#L346
and https://github.com/alexisvincent/systemjs-tools/blob/master/lib/index.js#L96.

**Review by alexisvincent — CHANGES_REQUESTED** [review]
Not happy here with the cache busting. Why do we need this?

**Comment by kwesterfeld** [comment]
The cache busting / eviction was required because I was getting stale data
that changed on the file system sent to the browser.   Perhaps it wasn't
the right location in the code to do this, but I needed it when modifying
.html/.css/.text assets.

I can revisit this now that the HMR issue was resolved with the other
changes.  I made the cache changes back in May so they are ancient. :)

**Comment by kwesterfeld** [comment]
To put a bit more color on the cache issue, the HMR path works fine.   I
change an .html file, it gets sent to the browser.  When I hit "refresh" on
the browser, however, the whole bundle is recompiled.  In this case, the
on-disk cache was serving assets that were cached prior to the changes
made, and thus you'd lose changes after a refresh.

**Comment by alexisvincent** [comment]
that sounds like a bug with `systemjs-builder` on file change I just call `builder.invalidate`... thoughts?

**Comment by alexisvincent** [comment]
is there a reason to believe the bug is with `systemjs-tools` and not the builder

**Comment by kwesterfeld** [comment]
I cannot argue with this logic.  When I was debugging the issue, I was less
cognizant of where the cache originated from, and merely changed the logic
there to evict the cache because I knew it was stale.

Probably the main issue is the cache key.  The changes I made are isolated
on this commit:
https://github.com/alexisvincent/systemjs-tools/pull/15/commits/4388c586b32f8fa43c6e4cf356593269fb82d6a7

**Comment by alexisvincent** [comment]
Got it :) Happy to merge if the cache commit is removed :) Thanks so much for the work you've done on this! Super nice to have all these issues fixed 👍 With a cool new feature as well!

**Comment by kwesterfeld** [comment]
Deal, I'll revert that commit and put it on a local branch, and work
through exactly what's going on.   I'll squash the other commits for a
single review

**Comment by kwesterfeld** [comment]
I squashed the sourcemaps and bundling customization changes for your review

**Comment by alexisvincent** [comment]
Awesome! thanks! merged

**Pull request merged**